sympy/simplify/hyperexpand.py

```diff
@@ -2304,7 +2306,7 @@ def do_slater(an, bm, ap, bq, z, zfinal):
                 # Now sum the finitely many residues:
                 # XXX This speeds up some cases - is it a good idea?
                 integrand = expand_func(integrand)
-                for r in range(lu):
+                for r in range(int(lu)):
```

Comment:
The parameters are sorted into "buckets" according to ``Mod(x, 1)`` so two any two in the same bucket differ by an integer. ``lu`` is the difference of two parameters in the same bucket.